### PR TITLE
Remove deprecated class map generator documentation

### PR DESCRIPTION
**Pull request by arueckauer**
This PR removes the documentation of the ZF 2 script `bin/classmap_generator.php`, which never was part of this component.

Instead the usage of Composer for autoloading is suggested and a reference to the migration guide added.

### CODE REVIEW

**Comment by froschdesign** [comment]
@weierophinney @michalbundyra
Is the `Zend\Loader\ClassMapAutoloader` deprecated or can this class be marked as deprecated?

**Comment by weierophinney** [comment]
Honestly, this entire package can likely be deprecated.

The only places it is used anywhere within ZF are:

- zend-modulemanager (which uses the `ModuleAutoloader` as a fallback to the application autoloader)
- zend-http (which uses the `PluginClassLoader` for loading HTTP header implementations, and should likely move to a standard plugin manager, but instead is going to be eventually deprecated in favor of Diactoros anyways)

In short: I don't know that we need to deprecate anything; it's more likely that we'll abandon it after the move to Laminas, in favor of inlining those facilities into the packages that use them currently.

**Comment by froschdesign** [comment]
@weierophinney 
> zend-modulemanager (which uses the `ModuleAutoloader` as a fallback to the application autoloader)

And there is already a pull request to drop zend-loader: https://github.com/zendframework/zend-modulemanager/pull/78 😃 

> …it's more likely that we'll abandon it after the move to Laminas…

Okay, that's the information we need here. Thanks!

**Review by froschdesign — CHANGES_REQUESTED** [review]
Only two minor changes are needed.

(We refuse all other adjustments here, like missing array example for `ClassMapAutoloader` or a separate migration section.)

File: docs/book/class-map-autoloader.md [hunk 1]
```diff
@@ -6,23 +6,14 @@ the file associated with that class on a match.  This avoids unnecessary
...
-The first step is to generate a class map file. You may run this over any
-directory containing source code anywhere underneath it.
-
-```bash
-$ php classmap_generator.php Some/Directory/
-```
-
-This will create a file named `Some/Directory/autoload_classmap.php`, which is a
-PHP file returning an associative array that represents the class map.
+The first step is to generate a class map file in form of a PHP file returning an
```
[the remark this answers is not in the document]
Comment (arueckauer) [reply docs/book/class-map-autoloader.md]: Done.

File: docs/book/classmap-generator.md [hunk 2]
```diff
@@ -1,28 +1,5 @@
 # The Class Map Generator utility: bin/classmap\_generator.php
 
-The script `bin/classmap_generator.php` can be used to generate class map files for use with the
-[ClassMapAutoloader](class-map-autoloader.md).
+The script `bin/classmap_generator.php` has been removed and is no longer available. Instead, we recommend using [Composer for autoloading](https://getcomposer.org/doc/04-schema.md#autoload).
```
[the remark this answers is not in the document]
Comment (arueckauer) [reply docs/book/classmap-generator.md]: Done.

**Review by froschdesign — APPROVED** [review]

**Comment by michalbundyra** [comment]
Thanks, @arueckauer!
